### DISCUSSION TITLE
Dialogflow And Webhooks

### DISCUSSION THREAD

**Discussion by leandroeldj** [chatwoot/chatwoot#10621] [Q&A]
Saludos comunidad de ChatWoot, queria saber si me pueden ayudar en algo por favor

Tengo configurado mi dialogflow perfectamente

![image](https://github.com/user-attachments/assets/35c98317-4ba8-443b-8c71-4d986107d68f)

Con telegram me funciona perfectamente
![image](https://github.com/user-attachments/assets/5ce86c65-d5ec-4708-ad63-105f6bf87478)

entonces e desarrollado un chat el cual comunico con chatwoot mediante WebHook y Api, e creado mi bandeja, pero cada vez que mando un mensaje a ChatWoot, la conversacion se asigna correctamente a dialogflow pero no me envia nada al Webhook, osea no recibo ningun tipo de repuesta, y ya le asigne dialogflow al canal correctamente pero aun no recibo nada en mi webhook

![image](https://github.com/user-attachments/assets/13b80afb-61c0-4db8-98ee-136773660f9d)

agradeceria mucho la ayuda, de ante mano muchas gracias

**Comment by CoderDeltaLAN** [comment]
Creo que aqui estas mezclando dos cosas distintas

Si usas la integracion nativa de Dialogflow en Chatwoot la respuesta del bot deberia volver por esa integracion, no por tu webhook externo

El webhook de Chatwoot sirve mas para avisarte de eventos de la conversacion, pero no es el mismo camino por donde responde el bot

Yo lo probaria por partes

Primero confirma que el inbox API crea la conversacion bien

Luego mira si Dialogflow esta asignado a ese inbox

Despues prueba una intencion simple tipo fallback para ver si el mensaje realmente llega a Dialogflow

Si lo que quieres es que tu backend responda directamente, quizas te conviene mas mirar Agent Bot en vez de la integracion nativa de Dialogflow

Yo no esperaria que Dialogflow responda a traves de tu webhook personalizado salvo que hayas montado tu esa logica completa
